### ISSUE TITLE
CodeMirror colorization -- how is it supposed to work?

### ISSUE THREAD

**Issue by robert-dodier** [fredokun/cl-jupyter#3]
Hi Frederic, how is CodeMirror supposed to work? I expected that when I enter some input, the input cell would be colorized, but that doesn't seem to be working for me. Does it work for you? (Have I misunderstood -- is the colorization supposed to be applied to output, not input?)

I haven't tried updating to cl-jupyter -- I am working with the older fishbowl-repl instead. I don't know if that makes a difference.

Thanks in advance for any insight!

**Comment by fredokun** [comment]
Dear Robert,  
Ipython2.x seems to use a relatively old codemirror without a good common lisp mode.  I tried to backport the lisp mode by directly loading it in the JS initialization ... but it does not seem to work and I don't know really why.    With Ipython 3.x  and now  Jupyter a more recent version of codemirror is used and  it is much simpler to enable the lisp mode (although I am still working on this because ipython keeps me sending non-UTF8 characters at initialization time and I don't know exactly why).

So I would suggest migrating to cl-jupyter to solve this particular issue.

**Comment by robert-dodier** [comment]
Hi Frederic, I was able to get the CodeMirror mode for Lisp to work by installing Jupyter and updating to cl-jupyter. There is a commonlisp.js in cl-jupyter but I believe that Jupyter installed a commonlisp.js and that is the one that is actually being used. 

I was also able to get Jupyter to use the CodeMirror mode for Maxima which I created by copying maxima.js to the system installation directory. I wasn't able to find any documentation about it, but I figured it out by running jupyter with the --debug option and seeing from where it was loading JS. I tried various ways to get jupyter to load maxima.js from a user directory but couldn't get anything to work.
